db/db_impl/db_impl_open.cc

```diff
@@ -340,6 +340,10 @@ Status DBImpl::NewDB(std::vector<std::string>* new_filenames) {
   }
   if (s.ok()) {
     // Make "CURRENT" file that points to the new manifest file.
+    s = directories_.SetDirectories(fs_.get(), dbname_,
+                                    immutable_db_options_.wal_dir,
+                                    immutable_db_options_.db_paths);
+    assert(s.ok());
```

Comment:
I checked `Recover()` does this before it calls `NewDB()`. Why is this needed?

Reply:
Yeah - it's a bit messy here and just for hack. See PR summary for the problem related to it.

Reply:
Do you mean this part: "it's because they are either just unit tests or internal logic where directories_ has not yet set properly with non-null db dir"? I don't know what is the code path where directories_ isn't set properly. DB logic looks ok thanks to https://github.com/facebook/rocksdb/blob/d17be55aab80b856f96f4af89f8d18fef96646b4/db/db_impl/db_impl_open.cc#L413-L415. Only other caller looks like repair tool, which seems able to set up directories_ if not already

Reply:
> Only other caller looks like repair tool, which seems able to set up directories_ if not already

This was hack that I used on my first draft to pass the all non-nullptr assertion for all tests. I can't remember exactly which UT now as we are removing that assertion and re-reading the code path seems fine with me without this hack.

Reply:
Fixed